core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -0,0 +1,943 @@
+    assertThat(view.history())
+        .hasSize(1)
+        .first()
+        .extracting(ViewHistoryEntry::versionId)
+        .isEqualTo(1);
+    assertThat(view.schema().schemaId()).isEqualTo(SCHEMA.schemaId());
+    assertThat(view.schema().asStruct()).isEqualTo(SCHEMA.asStruct());
+    assertThat(view.schemas()).hasSize(1).containsKey(SCHEMA.schemaId());
+    assertThat(view.versions()).hasSize(1).containsExactly(view.currentVersion());
+
+    assertThat(view.currentVersion())
+        .isEqualTo(
+            ImmutableViewVersion.builder()
+                .timestampMillis(view.currentVersion().timestampMillis())
+                .versionId(1)
+                .schemaId(SCHEMA.schemaId())
+                .putSummary("operation", "create")
+                .defaultNamespace(identifier.namespace())
+                .addRepresentations(
+                    ImmutableSQLViewRepresentation.builder()
+                        .sql("select * from ns.tbl")
+                        .dialect("spark")
+                        .build())
+                .addRepresentations(
+                    ImmutableSQLViewRepresentation.builder()
+                        .sql("select * from ns.tbl using X")
+                        .dialect("trino")
+                        .build())
+                .build());
+
+    assertThat(catalog().dropView(identifier)).isTrue();
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+  }
+
+  @Test
+  public void createViewThatAlreadyExists() {
+    TableIdentifier viewIdentifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(viewIdentifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(viewIdentifier)).isFalse();
+
+    View view =
+        catalog()
+            .buildView(viewIdentifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(viewIdentifier.namespace())
+            .withQuery("spark", "select * from ns.tbl")
+            .create();
+
+    assertThat(view).isNotNull();
+    assertThat(catalog().viewExists(viewIdentifier)).isTrue();
+
+    assertThatThrownBy(
+            () ->
+                catalog()
+                    .buildView(viewIdentifier)
+                    .withSchema(OTHER_SCHEMA)
+                    .withQuery("spark", "select * from ns.tbl")
+                    .withDefaultNamespace(viewIdentifier.namespace())
+                    .create())
+        .isInstanceOf(AlreadyExistsException.class)
+        .hasMessageStartingWith("View already exists: ns.view");
+  }
+
+  @Test
+  public void createViewThatAlreadyExistsAsTable() {
+    Assumptions.assumeThat(tableCatalog())
+        .as("Only valid for catalogs that support tables")
+        .isNotNull();
+
+    TableIdentifier tableIdentifier = TableIdentifier.of("ns", "table");
+    TableIdentifier viewIdentifier = TableIdentifier.of("ns", "view");
```

Comment:
This is again combining multiple test cases into a single method. I think that makes the tests less useful and much, much harder to read and understand. In this case, I missed the fact that there were two identifiers and you're exercising the view and table catalogs at the same time. Please separate test cases so that there are clear situations that are being tested. Duplication is much better than hard to read and maintain test cases, especially for SPI validation.

Reply:
sorry, my bad. I messed up conflict resolution during one of the rebases. You're right, this method shouldn't test what others are already testing and should have a clear purpose. I've slimmed down the test to only test what really matters here (creating a table and then making sure that you can't create a view with the same identifier)